### ISSUE TITLE
With().Str(...).Logger() fails while With().Timestamp().Str(...).Logger() works

### ISSUE THREAD

**Issue by fiatjaf** [rs/zerolog#16]
If I try to bind some parameters to a context without calling `Timestamp()` on it then it will fail at print time (I'm logging to Stdout with `zerolog.New(os.Stderr).Output(zerolog.ConsoleWriter{Out: os.Stderr})`).

The error is:

```
zerolog: could not write event: invalid character 'f' looking for beginning of object key string
```

`f` is the first character of the first bound parameter (the key). If I change the parameter the "invalid character" also changes.

**Comment by fiatjaf** [comment]
Smallest failing snippet:

```bash
fiatjaf@cantillon ~/c/test> cat x.go 
package main

import (
        "os"

        "github.com/rs/zerolog"
)

func main() {
        log := zerolog.New(os.Stderr).Output(zerolog.ConsoleWriter{Out: os.Stderr})

        bound := log.With().Str("x", "y").Logger()
        bound.Info().Msg("fails")

        bound = log.With().Timestamp().Str("x", "y").Logger()
        bound.Info().Msg("doesn't fail")
}
fiatjaf@cantillon ~/c/test> 
fiatjaf@cantillon ~/c/test> 
fiatjaf@cantillon ~/c/test> go run x.go 
zerolog: could not write event: invalid character 'x' looking for beginning of object key
string2017-09-25T16:53:07Z |INFO| doesn't fail x=y
```

**Comment by toravir** [comment]
Here the fix for this error - will create a pr with this fix shortly..
`
diff --git a/log.go b/log.go
index 83d1927..05a371e 100644
--- a/log.go
+++ b/log.go
@@ -166,7 +166,7 @@ func (l Logger) Output(w io.Writer) Logger {
 func (l Logger) With() Context {
        context := l.context
        l.context = make([]byte, 0, 500)
-       if context != nil {
+       if context != nil && len(context) != 0 {
                l.context = append(l.context, context...)
        } else {
                // first byte of context is presence of timestamp or not
`

**Comment by toravir** [comment]
I am sorry there is typo in this diff - it should be "if l.context != nil {" - NOT l2.context..

**Comment by rs** [comment]
Fixed by 9a65e7ccd2bad9082586a4de21f914648416c90f
